ecosystem/typescript/sdk/examples/typescript-esm/multi_ed25519_to_multisig.ts

```diff
@@ -0,0 +1,437 @@
+      BigInt(Math.floor(Date.now() / 1000) + secondsTilExpiration), // In an e2e flow, you may need to adjust this to account for the time it takes to gather the signatures.
+      new TxnBuilderTypes.ChainId(chainId),
+    );
+
+    const signingMessage = TransactionBuilder.getSigningMessage(rawTxn);
+    // The clients will each individually sign these from their own client in the e2e flow. The server gathers these hex string buffers later.
+    const rawTxnSignature1 = account1.signBuffer(signingMessage);
+    const rawTxnSignature2 = account2.signBuffer(signingMessage);
+    const rawTxnSignature3 = account3.signBuffer(signingMessage);
+
+    // Gather the signed hex buffers into Ed25519Signatures
+    const txnSig1 = new TxnBuilderTypes.Ed25519Signature(rawTxnSignature1.toUint8Array());
+    const txnSig2 = new TxnBuilderTypes.Ed25519Signature(rawTxnSignature2.toUint8Array());
+    const txnSig3 = new TxnBuilderTypes.Ed25519Signature(rawTxnSignature3.toUint8Array());
+
+    const multiEd25519Signature = new TxnBuilderTypes.MultiEd25519Signature([txnSig1, txnSig2, txnSig3], bitmap);
+
+    const multiEdAuthenticator = new TxnBuilderTypes.TransactionAuthenticatorMultiEd25519(
+      multiSigPublicKey,
+      multiEd25519Signature,
+    );
+    const signedTransaction = new TxnBuilderTypes.SignedTransaction(rawTxn, multiEdAuthenticator);
+    const bcsTxn = BCS.bcsToBytes(signedTransaction);
+
+    const transactionRes = await provider.submitSignedBCSTransaction(bcsTxn);
+
+    // Step 7.
+    // Wait for the transaction to complete, then observe and assert that the authentication key for the original MultiEd25519 account
+    // has been rotated and all capabilities revoked.
+    const txnInfo = await provider.waitForTransactionWithResult(transactionRes.hash);
+    printRelevantTxInfo(txnInfo as Types.UserTransaction);
+    assertChangesAndPrint(provider, multiSigAddress, sequenceNumber, true, metadataKeys, metadataValues);
+  }
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////////
+//                                                                                      //
+//                               Helper/utility functions                               //
+//                                                                                      //
+//////////////////////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////////
+
+// Funds and thus creates the derived MultiEd25519 account and prints out the derived address, authentication key, and public key.
+const initializeMultiEd25519 = async (
+  faucetClient: FaucetClient,
+  accounts: Array<AptosAccount>,
+  numSignaturesRequired: number,
+): Promise<Array<string>> => {
+  const multiSigPublicKey = new TxnBuilderTypes.MultiEd25519PublicKey(
```

Comment:
Let's add some comments here that creates an account with multiEd25519 auth key requires the public keys, which are different from account addresses. This has always been a very confusing detail for both devs and users as there's no easy way to get the public keys - some wallets don't really show them either.